src/workflows/airqo_etl_utils/satellite_data_utils.py

```diff
@@ -0,0 +1,229 @@
+
+                image_collection = ee.ImageCollection(image).select(['cloud_fraction', 'cloud_top_pressure'])
+                if image_collection.size().getInfo() == 0:
+                    print(f"No images available for {product}. Skipping.")
+                    continue
+
+                band_names = image_collection.first().bandNames().getInfo()
+
+                # Selection of appropriate bands and dates.
+                selected_bands = image_collection.select(ee.List(band_names)).filterDate(start_date, end_date)
+                if selected_bands.size().getInfo() == 0:
+                    print(f"No images available for {product} within the specified date range. Skipping.")
+                    continue
+
+                data = selected_bands.getRegion(site_location, 10).getInfo()
+                data_df = self.ee_array_to_df(data, band_names)
+                data_df.columns = [product + '_' + x for x in data_df.columns]
+                data_df['date'] = data_df[product + '_' + 'datetime'].map(lambda x: x.strftime('%Y-%m-%d'))
+                data_df['month'] = data_df[product + '_' + 'datetime'].dt.month
+                data_df['hour'] = data_df[product + '_' + 'datetime'].dt.hour
+                site_dfs[site] = data_df
+
+            dfs[product] = site_dfs
+
+        return dfs
+
+    def process_site_data(self, dfs):
+
+        site_dfs = {}
+        try:
+            for site in dfs[list(dfs.keys())[0]].keys():
+                products = []
+                # Iterate over each product
+                for product, site_df in dfs.items():
+                    # Retrieve dataframe for the current product and site
+                    product_df = site_df[site]
+                    product_df = product_df.groupby(['date', 'hour']).mean().reset_index()
+                    products.append(product_df)
+
+                # Merge dataframes for all products
+                merged_df = reduce(lambda left, right: pd.merge(left, right, on=['date', 'hour','month'], how='outer'), products)
+                merged_df['site_id'] = site
+                site_dfs[site] = merged_df
+
+        except Exception as e:
+            print(f"Error processing site data: {e}")
+
+        return site_dfs
+    
+
+    def merge_site_data(self, site_dfs):
+        result_df = pd.DataFrame()
+        for key in site_dfs.keys():
+          result_df = pd.concat([result_df, site_dfs[key]])
+        return result_df
+    
+    def extract_and_merge_data(self, df, result_df):
+      df1 = df[['site_id', 'site_name', 'pm2_5', 'pm10', 'date', 'site_latitude', 'site_longitude', 'country', 'month', 'hour']]
+      merged_df_ = df1.merge(result_df, how='right', on=['site_id', 'date', 'month', 'hour'])
+      merged_df_= merged_df_[~merged_df_['country'].isna()].reset_index(drop = True)
+      merged_df_ = merged_df_.drop(columns=[col for col in merged_df_.columns if '_datetime' in col])
+      merged_df_.to_csv('output.csv', index=False)
```

Comment:
Hi @wabinyai , why are you saving this CSV?
```suggestion

```

Reply:
It is for testing but not realy needed